Server/src/server/dataCenter/models/card/TempDeck.java

```diff
@@ -1,34 +1,9 @@
 package server.dataCenter.models.card;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-
-public class TempDeck {
-    private final String deckName;
-    private String heroId;
-    private final List<String> cardIds = new ArrayList<>();
+import shared.models.card.BaseTempDeck;
 
+public class TempDeck extends BaseTempDeck<ServerCard, Deck> {
     public TempDeck(Deck deck) {
-        this.deckName = deck.getName();
-        if (deck.getHero() != null) {
-            this.heroId = deck.getHero().getCardId();
-        }
-
-        for (ServerCard card : deck.getCards()) {
-            this.cardIds.add(card.getCardId());
-        }
-    }
-
-    public String getDeckName() {
-        return deckName;
-    }
-
-    String getHeroId() {
-        return heroId;
-    }
-
-    List<String> getCardIds() {
-        return Collections.unmodifiableList(cardIds);
+        super(deck);
```

Comment:
" if I wanted to access a method of the Deck class, I cant
more specifically, if you want to create a deck, the deck class calls the baseDeck's constructor, but the baseDeck constructor accepts DeckType as a parameter
If i want for example the players deck size, before I could easily do something like player.getDeck().getMinions().size()
But now I cant do that and I would have to find another way"